lib/dev/kit/system.rb

```diff
@@ -124,14 +124,17 @@ def system(*a, sudo: false, env: ENV, **kwargs)
               err_r => ->(data) { STDOUT.write(data) }, }
           end
 
+          previous_trailing = Hash.new('')
           loop do
             ios = [err_r, out_r].reject(&:closed?)
             break if ios.empty?
 
             readers, = IO.select(ios)
             readers.each do |io|
               begin
-                handlers[io].call(io.readpartial(4096))
+                data, trailing = split_partial_characters(io.readpartial(4096))
+                handlers[io].call(previous_trailing[io] + data)
```

Comment:
I don't love that we're copying the whole string again here, but I think this one can be considered minor, since we're at least not iterating over it.